packages/lodestar/src/chain/lightClient/index.ts

```diff
@@ -1 +1,579 @@
+   * Criteria for best in priority order:
+   * - Is finalized
+   * - Has the most bits
+   * - Signed header at the oldest slot
+   */
+  async serveBestUpdateInPeriod(period: SyncPeriod): Promise<altair.LightClientUpdate> {
+    // Signature data
+    const partialUpdate = await this.db.bestPartialLightClientUpdate.get(period);
+    if (!partialUpdate) {
+      throw Error(`No partialUpdate available for period ${period}`);
+    }
+
+    const syncCommitteeWitnessBlockRoot = partialUpdate.isFinalized
+      ? (partialUpdate.finalizedCheckpoint.root as Uint8Array)
+      : partialUpdate.blockRoot;
+
+    const syncCommitteeWitness = await this.db.syncCommitteeWitness.get(syncCommitteeWitnessBlockRoot);
+    if (!syncCommitteeWitness) {
+      throw Error(`finalizedBlockRoot not available ${toHexString(syncCommitteeWitnessBlockRoot)}`);
+    }
+
+    const nextSyncCommittee = await this.db.syncCommittee.get(syncCommitteeWitness.nextSyncCommitteeRoot);
+    if (!nextSyncCommittee) {
+      throw Error("nextSyncCommittee not available");
+    }
+
+    if (partialUpdate.isFinalized) {
+      return {
+        header: partialUpdate.finalizedHeader,
+        nextSyncCommittee: nextSyncCommittee,
+        nextSyncCommitteeBranch: getNextSyncCommitteeBranch(syncCommitteeWitness),
+        finalityHeader: partialUpdate.header,
+        finalityBranch: partialUpdate.finalityBranch,
+        syncCommitteeBits: partialUpdate.syncCommitteeBits,
+        syncCommitteeSignature: partialUpdate.syncCommitteeSignature,
+        forkVersion: this.config.getForkVersion(partialUpdate.header.slot),
+      };
+    } else {
+      return {
+        header: partialUpdate.header,
+        nextSyncCommittee: nextSyncCommittee,
+        nextSyncCommitteeBranch: getNextSyncCommitteeBranch(syncCommitteeWitness),
+        finalityHeader: this.zero.finalityHeader,
+        finalityBranch: this.zero.finalityBranch,
+        syncCommitteeBits: partialUpdate.syncCommitteeBits,
+        syncCommitteeSignature: partialUpdate.syncCommitteeSignature,
+        forkVersion: this.config.getForkVersion(partialUpdate.header.slot),
+      };
+    }
+  }
+
+  /**
+   * With forkchoice data compute which block roots will never become checkpoints and prune them.
+   */
+  async pruneNonCheckpointData(nonCheckpointBlockRoots: Uint8Array[]): Promise<void> {
+    // TODO: Batch delete with native leveldb batching not just Promise.all()
+    await Promise.all([
+      this.db.genesisWitness.batchDelete(nonCheckpointBlockRoots),
+      this.db.syncCommitteeWitness.batchDelete(nonCheckpointBlockRoots),
+      this.db.finalizedHeader.batchDelete(nonCheckpointBlockRoots),
+    ]);
+  }
+
+  private async persistPostBlockImportData(
+    block: altair.BeaconBlock,
+    postState: CachedBeaconState<allForks.BeaconState>,
+    parentBlock: {blockRoot: RootHex; slot: Slot}
+  ): Promise<void> {
+    const blockSlot = block.slot;
+
+    const header: phase0.BeaconBlockHeader = {
```

Comment:
we could use `blockToHeader()` util

Reply:
I've removed beacon-state-transition as dependency to make this lib lighter